test/OpenTelemetry.Tests/Logs/LoggerProviderTests.cs

```diff
@@ -0,0 +1,116 @@
+        }
+
+        [Fact]
+        public void VerifyResourceBuilderAddService()
+        {
+            var resource = CreateLoggerFactoryAndGetResource(options => options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(serviceName: "MyService", serviceVersion: "1.2.3")));
+
+            Assert.Contains(resource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString() == "MyService");
+            Assert.Contains(resource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceVersion && kvp.Value.ToString() == "1.2.3");
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithServiceNameEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, "MyService");
+
+                var resource = CreateLoggerFactoryAndGetResource();
+
+                Assert.Contains(resource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString() == "MyService");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, null);
+            }
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithAttributesEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, "Key1=Val1,Key2=Val2");
+
+                var resource = CreateLoggerFactoryAndGetResource();
+
+                Assert.Contains(resource.Attributes, (kvp) => kvp.Key == "Key1" && kvp.Value.ToString() == "Val1");
+                Assert.Contains(resource.Attributes, (kvp) => kvp.Key == "Key2" && kvp.Value.ToString() == "Val2");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, null);
+            }
+        }
+
+        private static Resource CreateLoggerFactoryAndGetResource(Action<OpenTelemetryLoggerOptions> configure = null)
+        {
+            using var exporter = new InMemoryExporter<LogRecord>(new List<LogRecord>());
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    configure?.Invoke(options);
+                    options.AddProcessor(new TestLogRecordProcessor(exporter));
+                });
+                builder.AddFilter(typeof(LoggerProviderTests).FullName, LogLevel.Trace);
+            });
+            var logger = loggerFactory.CreateLogger<LoggerProviderTests>();
+
+            var provider = exporter.ParentProvider as OpenTelemetryLoggerProvider;
+            return provider.GetResource();
```

Comment:
it feels a bit odd to continue using Resource, *after* disposing factory (and therefore provider)... Could this helper method take an action, containing the validation logic as well, so that the validation can also occur right here, before anything gets disposed?

Reply:
hmmmmm, yes that would be better. i'll make the change.